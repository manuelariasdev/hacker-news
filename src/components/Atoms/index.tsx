import { Ensign } from './Ensign'
import { Button } from './Button'
import { Icon } from './Icon'
import { IconButton } from './IconButton'
import { Link } from './Link'
import { Carcass } from './Carcass'
import { Spinner } from './Spinner'
import { Text } from './Text'
import { Title } from './Title'


export {
  Ensign,
  Button,
  Icon,
  IconButton,
  Link,
  Carcass,
  Spinner,
  Text,
  Title
}
