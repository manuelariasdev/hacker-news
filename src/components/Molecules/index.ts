import { Card } from './Card'
import { Flex } from './Flex'
import { List } from './List'
import { DropDown } from './DropDown'
import { Pagination } from './Pagination';

export {Card, Flex, List, DropDown, Pagination  }
