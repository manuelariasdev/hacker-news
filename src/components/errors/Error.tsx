import React from 'react'

export const Error = () => {
  return (
    <div><h1>BAD BAD BAD BAD BAD</h1></div>
  )
}
