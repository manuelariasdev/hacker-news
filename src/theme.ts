
import { DefaultTheme } from 'styled-components'

const theme: DefaultTheme = {
  borderRadius: {
    sm: '3px',
    md: '6px',
    lg: '15px',
    xl: '50px'
  },
  boxShadow: {
    sm: ' 0 1px 4px 0 rgba(0, 21, 41, 0.12)'
  },
  fontSize: {
    xs: '0.68rem', 
    sm: '0.875rem', 
    md: '16px',
    lg: '1.25rem', 
    xl: '1.75rem'
  },
  letterSpacing: {
    md: 0.25
  },
  lineHeight: {
    xs: '1rem',
    sm: '1.25rem',
    md: '16px',
    lg: '1.25rem',
    xl: '1.75rem'
  },
  transition: 'all 700ms cubic-bezier(0.15, 1, 0.20, 1)'
}

export default theme
