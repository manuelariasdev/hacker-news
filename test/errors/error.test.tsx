/**
 * @jest-environment jsdom
 */
import * as React from "react";
import { ThemeProvider } from 'styled-components'
import { render, screen } from '@testing-library/react'
import theme from '../../theme';
import { Error } from '../../src/components/errors'
describe('Error <Error/>', function () {

    test('debe de hacer match con el snapshot Error', () => {
        const {container} = render(<ThemeProvider theme={theme}><Error /> </ThemeProvider>)
        expect( container ).toMatchSnapshot();
     })

     test('debe de mostrar BAD', () => {

        render(<ThemeProvider theme={theme}><Error /> </ThemeProvider>)
        expect(screen.getByText('BAD BAD BAD BAD BAD')).toBeTruthy();
        
     })
  

});