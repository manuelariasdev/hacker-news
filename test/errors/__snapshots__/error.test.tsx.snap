// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error <Error/> debe de hacer match con el snapshot Error 1`] = `
<div>
  <div>
    <h1>
      BAD BAD BAD BAD BAD
    </h1>
  </div>
   
</div>
`;
